code/modules/mob/living/silicon/pai/software_modules.dm

```diff
@@ -354,11 +354,11 @@
 /mob/living/silicon/pai/proc/hackloop()
 	var/turf/T = get_turf_or_move(src.loc)
 	for(var/mob/living/silicon/ai/AI in player_list)
-		if(T.loc)
+		if(T.loc && prob(60)) //60% chance that AI notices hack
```

Comment:
As discussed on discord: Remove the chance. 
AI should always get a alert if someone tries to hack a airlock with a pAI.